Modules/_sqlite/connection.c

```diff
@@ -107,6 +106,21 @@ new_statement_cache(pysqlite_Connection *self, int maxsize)
     return res;
 }
 
+static const char *
+get_begin_statement(const char *level, Py_ssize_t sz)
+{
+    assert(level != NULL);
+    for (int i = 0; begin_statements[i] != NULL; i++) {
+        const char *candidate = begin_statements[i] + 6;
+        if (strlen(candidate) == (size_t)sz) {
```

Comment:
Do we really need a length check here?

Reply:
Yes, unfortunately. There are some corner cases with embedded null characters that will fail. The test suite already got regression tests for these.

Reply:
What I could do is to check for embedded null characters in `pysqlite_connection_set_isolation_level`. Then we can remove the strlen check (and the size argument). Might be a better solution.

Reply:
I pushed this change; I think it makes for more readable code:

- `get_begin_statement()` is clearer
- there is no need to call `strlen()` in `__init__`